pgmax/fg/graph.py

```diff
@@ -146,7 +151,7 @@ def add_factor_by_type(
             To add an ORFactor to a FactorGraph fg, run::
 
                 fg.add_factor_by_type(
-                    variable_names=variables_names_for_OR_factor,
+                    variable_names_for_factors=[variables_names_for_OR_factor],
```

Comment:
Why this change? This function is meant to add single factors imperatively, so there should only be one set of `variable_names`

Reply:
I see it's to make all factor groups have the same consistent interface. That is fine, but to avoid confusion, the argument for `add_factor_by_type` should still be just `variable_names`. Inside the function we can wrap it inside a list and pass the list to `SingleFactorGroup`.

After the change `SingleFactorGroup` should have an additional check to make sure its `len(variable_names_for_factors) == 1`.

Reply:
totally agree